Update Haskell package set to Stackage Nightly 2021-01-20 (plus other fixes)

peti opened the pull request:
This PR is test-built by Hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. I'll fix up the remaining errors and merge it on Friday, 2021-01-22 20:00 +01:00. You can watch this live on Twitch at https://www.twitch.tv/peti343.

peti committed:
Stackage Nightly 2021-01-20

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.6 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/8c276aee32a6348165828e74e53b81a184e4e577.

maralorn commented:
pandoc broke this night. I have reported it upstream: https://github.com/jgm/pandoc/issues/7035

maralorn committed:
haskellPackages.pandoc: Temporarily disable tests

peti committed:
hackage2nix: update list of broken builds to fix evaluation errors on Hydra

peti committed:
hnix-store-core: add missing test dependency by way of override

See https://github.com/haskell-nix/hnix-store/issues/127.

peti committed:
hnix-store-core: update overrides for the latest version

maralorn committed:
haskellPackages.universe-dependent-sum: Remove broken flag

peti committed:
Consistently use 'lib' instead of 'stdenv.lib' in the Haskell infrastructure.

Part of https://github.com/NixOS/nixpkgs/issues/108938.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.16.0-9-ga573266 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/61c759fc797fec930eaed15016faa4bdf7c53c75.

maralorn committed:
haskellPackages.haskell-language-server: Fixing build

by pinning apply-refact

Prillan committed:
haskellPackages.glicko: unbreak

I'm the author of the package and fixed the dependencies upstream,
now builds correctly and can be unmarked as broken.

peti commented:
We'll merge https://github.com/NixOS/nixpkgs/pull/111156 next week.
